pkg/reconciler/pipeline/validate.go

```diff
@@ -89,6 +89,17 @@ func ValidatePipeline(pl *dfv1.Pipeline) error {
 		return fmt.Errorf("pipeline has no sink, at least one vertex with 'sink' defined is required")
 	}
 
+	for _, v := range pl.Spec.Vertices {
```

Comment:
Shall we do something like below:

1. Sources should have `to`, but no `from`,
2. Sinks should have `from`, but no`to`;
3. UDFs should have both.

And this can be done together with the for loop above.

Reply:
If this logic is in place, maybe we can remove the logic in line 165-170.

Reply:
+1.